lib/rpmscript.c

```diff
@@ -103,42 +103,28 @@ static rpmRC runLuaScript(rpmPlugins plugins, ARGV_const_t prefixes,
 {
     rpmRC rc = RPMRC_FAIL;
 #ifdef WITH_LUA
-    ARGV_t argv = argvp ? *argvp : NULL;
     rpmlua lua = NULL; /* Global state. */
-    rpmluav var = rpmluavNew();
     int cwd = -1;
 
     rpmlog(RPMLOG_DEBUG, "%s: running <lua> scriptlet.\n", sname);
 
-    /* Create arg variable */
-    rpmluaPushTable(lua, "arg");
-    rpmluavSetListMode(var, 1);
     rpmluaSetNextFileFunc(nextFileFunc->func, nextFileFunc->param);
-    if (argv) {
-	char **p;
-	for (p = argv; *p; p++) {
-	    rpmluavSetValue(var, RPMLUAV_STRING, *p);
-	    rpmluaSetVar(lua, var);
-	}
-    }
-    if (arg1 >= 0) {
-	rpmluavSetValueNum(var, arg1);
-	rpmluaSetVar(lua, var);
-    }
-    if (arg2 >= 0) {
-	rpmluavSetValueNum(var, arg2);
-	rpmluaSetVar(lua, var);
-    }
-    rpmluaPop(lua);
+
+    if (arg1 >= 0)
+	argvAddNum(argvp, arg1);
+    if (arg2 >= 0)
+	argvAddNum(argvp, arg2);
 
     /* Lua scripts can change our cwd and umask, save and restore */
     cwd = open(".", O_RDONLY);
     if (cwd != -1) {
 	mode_t oldmask = umask(0);
 	umask(oldmask);
 	pid_t pid = getpid();
+	/* Arrange arguments to appear as a local arg[n] table in the script */
+	char *s = rstrscat(NULL, argvp ? "arg = {...}\n" : "", script, NULL);
```

Comment:
argvp can not be NULL so you might want to drop the test. (The rest of the code already assumes that it is not NULL.)

Reply:
Oh, right. This snippet was originally in rpmluaRunScript() where it could've been NULL so it was necessary there but not here. Thanks for spotting.